src/rust/ide/view/src/documentation.rs

```diff
@@ -133,22 +148,42 @@ impl ViewModel {
             //              https://github.com/enso-org/enso/issues/1063
             let processed = string.replace("\\n", "\n").replace("\"", "");
             let output = match input_type {
-                InputFormat::AST  => parser.generate_html_docs(processed),
+                InputFormat::AST       => parser.generate_html_docs(processed),
                 InputFormat::Docstring => parser.generate_html_doc_pure(processed),
             };
             let output = output?;
-            if output.is_empty() {
-                Ok(PLACEHOLDER_STR.into())
-            } else {
-                Ok(output)
-            }
+            if output.is_empty() { Ok(PLACEHOLDER_STR.into()) }
+            else                 { Ok(output)                 }
         }
     }
 
     /// Create a container for generated content and embed it with stylesheet.
     fn push_to_dom(&self, content:String) {
         let data_str = format!(r#"<div class="docVis">{}{}</div>"#,documentation_style(),content);
-        self.dom.dom().set_inner_html(&data_str)
+        self.dom.dom().set_inner_html(&data_str);
+    }
+
+    /// Append listeners to copy buttons in doc to enable copying examples.
+    fn set_listeners_to_copy_buttons(&self) {
+        let code_blocks  = self.dom.dom().get_elements_by_class_name("CodeBlock");
+        let copy_buttons = self.dom.dom().get_elements_by_class_name("copyCodeBtn");
+        let closures     = (0..copy_buttons.length()).map(|i| -> OptClosure {
+            let copy_button = copy_buttons.get_with_index(i)?.dyn_into::<HtmlElement>().ok()?;
+            let code_block  = code_blocks.get_with_index(i)?.dyn_into::<HtmlElement>().ok()?;
+            println!("{:?}",&code_block.inner_text());
+            let closure  = move |_event: MouseEvent| {
+                let inner_code = code_block.inner_text();
+                clipboard::write_text(inner_code);
+            };
+            let closure  = Closure::wrap(Box::new(closure).into_fn_mut());
+            let callback = closure.as_ref().unchecked_ref();
+            match copy_button.add_event_listener_with_callback("click", callback){
+                Ok(_)    => (),
+                Err(err) => error!(&self.logger, "Unable to add event listener: {err:?}")
+            }
+            Some(closure)
+        }).collect::<Vec<OptClosure>>();
```

Comment:
Why you are collecting opt closures instead of collecting only closures and skipping all Nones? I don't think that collecting Nones is good idea here.